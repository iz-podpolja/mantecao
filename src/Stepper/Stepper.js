import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { ArrowIcon } from '../icons'

import {
  StyledDiv,
  StyledStepperButton,
  StyledExpander,
  StyledControls,
  StyledChildren
} from './_Styled'

const Stepper = ({
  children,
  icon,
  increaseIcon,
  decreaseIcon,
  disabled,
  onClick,
  onDecreaseClick,
  onIncreaseClick
}) => {
  return (
    <StyledDiv disabled={disabled}>

      <StyledChildren onClick={disabled ? () => false : onClick}>
        {children}
      </StyledChildren>

      <StyledExpander
        disabled={disabled}
        onClick={disabled ? () => false : onClick}
      >
        {icon ||
          <ArrowIcon
            width="1em"
            height="1em"
            style={{ strokeWidth: '2.5px' }}
          />}
      </StyledExpander>

      <StyledControls>
        <StyledStepperButton onClick={onIncreaseClick} disabled={disabled}>
          {increaseIcon ||
            <ArrowIcon
              width="1em"
              height="1em"
              style={{ strokeWidth: '2.5px' }}
              mirrorX
            />}
        </StyledStepperButton>
        <StyledStepperButton onClick={onDecreaseClick} disabled={disabled}>
          {increaseIcon ||
            <ArrowIcon
              width="1em"
              height="1em"
              style={{ strokeWidth: '2.5px' }}
            />}
        </StyledStepperButton>
      </StyledControls>

    </StyledDiv>
  )
}

Stepper.PropTypes = {
  children: PropTypes.node.isRequired,
  decreaseIcon: PropTypes.node,
  disabled: PropTypes.bool,
  icon: PropTypes.node,
  increaseIcon: PropTypes.node,
  onClick: PropTypes.function,
  onDecreaseClick: PropTypes.function,
  onIncreaseClick: PropTypes.function
}

export default Stepper
