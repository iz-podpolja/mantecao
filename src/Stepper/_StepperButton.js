import React from 'react'

const StepperButton = ({ id, action, disabled, icon }) => {
  return (
    <span>
      {icon}
    </span>
  )
}

export default StepperButton
