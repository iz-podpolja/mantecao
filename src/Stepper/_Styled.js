import React from 'react'
import styled from 'styled-components'

import { colors, theme } from '../variables'

export const StyledStepperButton = styled.button`
  background-color: ${colors.white};
  border-top: none;
  border-right: none;
  border-bottom: none;
  border-radius: 0;
  border-left: 1px solid currentColor;
  color: currentColor;
  cursor: pointer;
  display: flex;
  flex: 1 1 auto;
  flex-direction: column;
  font-size: ${theme.fontSize};
  justify-content: center;
  opacity: 1;
  padding: ${theme.smallSpace} ${theme.baseSpace};
  white-space: normal;

  &:hover,
  &:focus,
  &:disabled,
  &:disabled:hover {
    background-color: ${colors.whiteSand};
    color: currentColor;
    opacity: 1;
  }

  &:first-of-type {
    border-top-right-radius: ${theme.borderRadius};
    border-bottom: 1px solid currentColor;
  }

  &:last-of-type {
    border-bottom-right-radius: ${theme.borderRadius};
  }
`

export const StyledDiv = styled.div`
  border: 1px solid ${colors.pacificBlue};
  border-radius: ${theme.borderRadius};
  color: ${theme.fontColor};
  cursor: ${props => (props.disabled ? 'not-allowed' : 'inherit')};
  display: flex;
  font-size: ${theme.fontSize};
  opacity: ${props => (props.disabled ? 0.5 : 1)};
  padding-left: ${theme.baseSpace};
`

export const StyledChildren = styled.div`
  display: flex;
  flex: 1 1 auto;
  flex-direction: column;
  justify-content: center;
`

export const StyledExpander = styled.div`
  color: ${colors.silver};
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  display: flex;
  flex: 0 0 auto;
  flex-direction: column;
  justify-content: center;
  padding: ${theme.baseSpace};
`

export const StyledControls = styled.div`
  color: ${colors.silver};
  display: flex;
  flex: 0 0 auto;
  flex-direction: column;
  justify-content: center;
`
