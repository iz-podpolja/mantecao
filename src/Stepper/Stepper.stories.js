import React from 'react'
import { storiesOf } from '@storybook/react'
import Stepper from './Stepper'

storiesOf('Stepper', module).add('default', () => <Stepper>Hello 💩</Stepper>)
