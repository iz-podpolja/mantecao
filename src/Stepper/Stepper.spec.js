import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Stepper from './Stepper'

const ON_CLICK = jest.fn()
const INCREASE_ACTION = jest.fn()
const DECREASE_ACTION = jest.fn()

describe('Stepper', () => {
  let stepper

  beforeEach(() => {
    stepper = shallow(
      <Stepper
        onClickAction={ON_CLICK}
        onIncreaseClick={INCREASE_ACTION}
        onDecreaseClick={DECREASE_ACTION}
      >
        Click
      </Stepper>
    )
  })

  xit('Renders its children', () => {
    const tree = renderer
      .create(<Stepper>👶👶🏻👶🏼👶🏽👶🏾👶🏿</Stepper>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Should run the OnClickAction passed as a prop when clicked', () => {
    stepper.simulate('click')
    expect(ON_CLICK).toHaveBeenCalled()
  })

  it('Should run the onIncreaseClick action when the inc button is clicked ', () => {
    stepper.find('#mantecao-stepper-inc').simulate('click')
    expect(INCREASE_ACTION).toBeCalled()
  })

  it('Should run the onDecreaseClick action when the inc button is clicked ', () => {
    stepper.find('#mantecao-stepper-dec').simulate('click')
    expect(DECREASE_ACTION).toBeCalled()
  })

  it('Should not run the onClickAction action when the inc button is clicked ', () => {
    ON_CLICK.mockClear()
    stepper.find('#mantecao-stepper-inc').simulate('click')
    expect(ON_CLICK).not.toBeCalled()
  })
})
