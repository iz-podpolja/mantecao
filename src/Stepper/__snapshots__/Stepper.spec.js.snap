// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper renders custom icons 1`] = `
<div
  className="sc-bwzfXH dqIYPn"
  disabled={undefined}
>
  <div
    className="sc-htpNat cCQUd"
    onClick={undefined}
  />
  <div
    className="sc-bxivhb hHsYUR"
    disabled={undefined}
    onClick={undefined}
  >
    <svg
      className={undefined}
      style={
        Object {
          "display": "inline-block",
          "fill": "none",
          "height": "2.1rem",
          "pointerEvents": "none",
          "stroke": "currentColor",
          "strokeWidth": 1.5,
          "verticalAlign": "middle",
          "width": "2.1rem",
        }
      }
      viewBox="0 0 32 32"
    >
      <g
        transform={undefined}
      >
        <g
          transform="translate(1 8)"
        >
          <path
            d="M14.806 16.59c8.686 0 14.133-6.603 14.133-6.603.37-.403.375-1.06-.014-1.467 0 0-5.434-6.54-14.12-6.54C6.12 1.98.674 8.532.674 8.532c-.372.4-.376 1.06 0 1.454 0 0 5.447 6.602 14.132 6.602zm.194-.026c4.01 0 7.258-3.272 7.258-7.308S19.008 1.95 15 1.95c-4.01 0-7.258 3.27-7.258 7.306s3.25 7.308 7.258 7.308zm-9.92 2.192L24.92.244"
            stroke="currentColor"
            strokeWidth="1.3"
          />
          <ellipse
            cx="14.758"
            cy="9.5"
            fill="currentColor"
            rx="3.629"
            ry="3.654"
          />
        </g>
      </g>
    </svg>
  </div>
  <div
    className="sc-ifAKCX bCKPal"
  >
    <button
      className="sc-bdVaJa fzYoTq"
      disabled={undefined}
      onClick={undefined}
    >
      <svg
        className={undefined}
        style={
          Object {
            "display": "inline-block",
            "fill": "none",
            "height": "2.1rem",
            "pointerEvents": "none",
            "stroke": "currentColor",
            "strokeWidth": 1.5,
            "verticalAlign": "middle",
            "width": "2.1rem",
          }
        }
        viewBox="0 0 32 32"
      >
        <g
          transform={undefined}
        >
          <g
            transform="translate(1 8)"
          >
            <path
              d="M14.806 16.59c8.686 0 14.133-6.603 14.133-6.603.37-.403.375-1.06-.014-1.467 0 0-5.434-6.54-14.12-6.54C6.12 1.98.674 8.532.674 8.532c-.372.4-.376 1.06 0 1.454 0 0 5.447 6.602 14.132 6.602zm.194-.026c4.01 0 7.258-3.272 7.258-7.308S19.008 1.95 15 1.95c-4.01 0-7.258 3.27-7.258 7.306s3.25 7.308 7.258 7.308zm-9.92 2.192L24.92.244"
              stroke="currentColor"
              strokeWidth="1.3"
            />
            <ellipse
              cx="14.758"
              cy="9.5"
              fill="currentColor"
              rx="3.629"
              ry="3.654"
            />
          </g>
        </g>
      </svg>
    </button>
    <button
      className="sc-bdVaJa fzYoTq"
      disabled={undefined}
      onClick={undefined}
    >
      <svg
        className={undefined}
        style={
          Object {
            "display": "inline-block",
            "fill": "none",
            "height": "2.1rem",
            "pointerEvents": "none",
            "stroke": "currentColor",
            "strokeWidth": 1.5,
            "verticalAlign": "middle",
            "width": "2.1rem",
          }
        }
        viewBox="0 0 32 32"
      >
        <g
          transform={undefined}
        >
          <g
            transform="translate(1 8)"
          >
            <path
              d="M14.806 16.59c8.686 0 14.133-6.603 14.133-6.603.37-.403.375-1.06-.014-1.467 0 0-5.434-6.54-14.12-6.54C6.12 1.98.674 8.532.674 8.532c-.372.4-.376 1.06 0 1.454 0 0 5.447 6.602 14.132 6.602zm.194-.026c4.01 0 7.258-3.272 7.258-7.308S19.008 1.95 15 1.95c-4.01 0-7.258 3.27-7.258 7.306s3.25 7.308 7.258 7.308zm-9.92 2.192L24.92.244"
              stroke="currentColor"
              strokeWidth="1.3"
            />
            <ellipse
              cx="14.758"
              cy="9.5"
              fill="currentColor"
              rx="3.629"
              ry="3.654"
            />
          </g>
        </g>
      </svg>
    </button>
  </div>
</div>
`;

exports[`Stepper renders its children 1`] = `
<div
  className="sc-bwzfXH dqIYPn"
  disabled={undefined}
>
  <div
    className="sc-htpNat cCQUd"
    onClick={undefined}
  >
    👶👶🏻👶🏼👶🏽👶🏾👶🏿
  </div>
  <div
    className="sc-bxivhb hHsYUR"
    disabled={undefined}
    onClick={undefined}
  >
    <svg
      className={undefined}
      style={
        Object {
          "display": "inline-block",
          "fill": "none",
          "height": "1em",
          "pointerEvents": "none",
          "stroke": "currentColor",
          "strokeWidth": "2.5px",
          "verticalAlign": "middle",
          "width": "1em",
        }
      }
      viewBox="0 0 32 32"
    >
      <g
        transform={undefined}
      >
        <g
          transform="translate(0.75, -1)"
        >
          <path
            d="M3 12.1L14.6 24.5C15 24.9 15.6 24.9 16 24.5L27.5 12"
          />
        </g>
      </g>
    </svg>
  </div>
  <div
    className="sc-ifAKCX bCKPal"
  >
    <button
      className="sc-bdVaJa fzYoTq"
      disabled={undefined}
      onClick={undefined}
    >
      <svg
        className={undefined}
        style={
          Object {
            "display": "inline-block",
            "fill": "none",
            "height": "1em",
            "pointerEvents": "none",
            "stroke": "currentColor",
            "strokeWidth": "2.5px",
            "verticalAlign": "middle",
            "width": "1em",
          }
        }
        viewBox="0 0 32 32"
      >
        <g
          transform="translate(0, 32) scale(1, -1)"
        >
          <g
            transform="translate(0.75, -1)"
          >
            <path
              d="M3 12.1L14.6 24.5C15 24.9 15.6 24.9 16 24.5L27.5 12"
            />
          </g>
        </g>
      </svg>
    </button>
    <button
      className="sc-bdVaJa fzYoTq"
      disabled={undefined}
      onClick={undefined}
    >
      <svg
        className={undefined}
        style={
          Object {
            "display": "inline-block",
            "fill": "none",
            "height": "1em",
            "pointerEvents": "none",
            "stroke": "currentColor",
            "strokeWidth": "2.5px",
            "verticalAlign": "middle",
            "width": "1em",
          }
        }
        viewBox="0 0 32 32"
      >
        <g
          transform={undefined}
        >
          <g
            transform="translate(0.75, -1)"
          >
            <path
              d="M3 12.1L14.6 24.5C15 24.9 15.6 24.9 16 24.5L27.5 12"
            />
          </g>
        </g>
      </svg>
    </button>
  </div>
</div>
`;

exports[`Stepper renders with click handlers 1`] = `
<div
  className="sc-bwzfXH dqIYPn"
  disabled={undefined}
>
  <div
    className="sc-htpNat cCQUd"
    onClick={[Function]}
  />
  <div
    className="sc-bxivhb hHsYUR"
    disabled={undefined}
    onClick={[Function]}
  >
    <svg
      className={undefined}
      style={
        Object {
          "display": "inline-block",
          "fill": "none",
          "height": "1em",
          "pointerEvents": "none",
          "stroke": "currentColor",
          "strokeWidth": "2.5px",
          "verticalAlign": "middle",
          "width": "1em",
        }
      }
      viewBox="0 0 32 32"
    >
      <g
        transform={undefined}
      >
        <g
          transform="translate(0.75, -1)"
        >
          <path
            d="M3 12.1L14.6 24.5C15 24.9 15.6 24.9 16 24.5L27.5 12"
          />
        </g>
      </g>
    </svg>
  </div>
  <div
    className="sc-ifAKCX bCKPal"
  >
    <button
      className="sc-bdVaJa fzYoTq"
      disabled={undefined}
      onClick={[Function]}
    >
      <svg
        className={undefined}
        style={
          Object {
            "display": "inline-block",
            "fill": "none",
            "height": "1em",
            "pointerEvents": "none",
            "stroke": "currentColor",
            "strokeWidth": "2.5px",
            "verticalAlign": "middle",
            "width": "1em",
          }
        }
        viewBox="0 0 32 32"
      >
        <g
          transform="translate(0, 32) scale(1, -1)"
        >
          <g
            transform="translate(0.75, -1)"
          >
            <path
              d="M3 12.1L14.6 24.5C15 24.9 15.6 24.9 16 24.5L27.5 12"
            />
          </g>
        </g>
      </svg>
    </button>
    <button
      className="sc-bdVaJa fzYoTq"
      disabled={undefined}
      onClick={[Function]}
    >
      <svg
        className={undefined}
        style={
          Object {
            "display": "inline-block",
            "fill": "none",
            "height": "1em",
            "pointerEvents": "none",
            "stroke": "currentColor",
            "strokeWidth": "2.5px",
            "verticalAlign": "middle",
            "width": "1em",
          }
        }
        viewBox="0 0 32 32"
      >
        <g
          transform={undefined}
        >
          <g
            transform="translate(0.75, -1)"
          >
            <path
              d="M3 12.1L14.6 24.5C15 24.9 15.6 24.9 16 24.5L27.5 12"
            />
          </g>
        </g>
      </svg>
    </button>
  </div>
</div>
`;
